apps/test/unit/templates/instructions/TopInstructionsCSPTest.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../../util/configuredChai';
+import {UnconnectedTopInstructionsCSP as TopInstructionsCSP} from '@cdo/apps/templates/instructions/TopInstructionsCSP';
+//import {stubRedux, restoreRedux, getStore} from '@cdo/apps/redux';
+//import {Provider} from 'react-redux';
+
+const DEFAULT_PROPS = {
+  isEmbedView: false,
+  hasContainedLevels: false,
+  puzzleNumber: 2,
+  stageTotal: 0,
+  height: 200,
+  expandedHeight: 300,
+  maxHeight: 300,
+  longInstructions: 'Some instructions for the level',
+  collapsed: false,
+  noVisualization: false,
+  toggleInstructionsCollapsed: () => {},
+  setInstructionsHeight: () => {},
+  setInstructionsRenderedHeight: () => {},
+  setInstructionsMaxHeightNeeded: () => {},
+  viewAs: 'Teacher',
+  readOnlyWorkspace: false,
+  serverLevelId: 123,
+  user: 5
+};
+
+describe('TopInstructionsCSP', () => {
+  describe('Feedback Tab', () => {
+    it('While giving feedback I switch to the instructions tab and back', () => {
+      /* const wrapper = shallow(<TopInstructionsCSP {...DEFAULT_PROPS} />);
+
+      wrapper.setState({
+        tabSelected: 'comments',
+        feedbacks: [],
+        rubric: null,
+        displayFeedbackTeacherFacing: true
+      });
+
+      const instructionsTab = wrapper.find('.uitest-instructionsTab').at(0);
+      instructionsTab.simulate('click');
+      expect(
+        wrapper.find('TeacherFeedback').props().visible
+      ).to.equal(false);
+
+      const feedbackTab = wrapper.find('.uitest-feedback').at(0);
+      feedbackTab.simulate('click');
+      expect(
+        wrapper.find('TeacherFeedback').props().visible
+      ).to.equal(true);
+      */
+    });
+    it('height of instructions area is full size needed for content when feedback tab opened', () => {
+      /*
+      stubRedux();
+      const store = getStore();
+      const wrapper = mount(
+        <Provider store={store}>
+          <TopInstructionsCSP {...DEFAULT_PROPS} />
+        </Provider>
+      );
+
+      wrapper.setState({
+        tabSelected:'comments',
+        feedbacks: [],
+        rubric: null,
+        displayFeedbackTeacherFacing: false
+      });
+
+      expect(wrapper.props().height).to.equal(wrapper.props().maxHeight);
```

Comment:
Where does the actual height logic live? I wonder how much we can isolate it and test it without React at all.

Reply:
Height logic lives in instructions redux and is tested in instructionsTest. We do have eyes tests that are watching some of this. Maybe that's good enough?

Reply:
Yeah, it might be. It's a little tricky for this to live in, and be tested in, redux, given it's a feature so fundamentally tied to DOM state.  I'm also worried that the regressions we've seen in this area recently suggests that the existing test coverage isn't doing a good job of catching tests.  What are your thoughts on that?

It's possible we need some serious refactoring to make this unit-testable or even headless-integration-testable.  Or, maybe our eyes test just needs to be extended to cover more scenarios.  Either way I think you could consider that problem as separate from this PR.

Reply:
I think adding more eyes test around this might be the easiest way to test for right now considering how messy the height code is for the instructions area. I think a good tech debt for the future would be to clean up the height of instructions area code to make it less complicated and more testable